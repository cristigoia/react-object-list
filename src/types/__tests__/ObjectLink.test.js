import React from 'react'
import { shallow } from 'enzyme'
import {
  snapshotTest,
} from 'utils/tests'
import { ObjectLink, Links } from '../'

describe('ObjectLink type', () => {
  it('snapshot tests', () => {
    const input = {data: null}
    const noAttrs = {data: {id: 1, type: 'TestModel'}}
    const noLink = {...noAttrs,
      attributes: {
        __str__: 'default',
      },
    }
    const methodLink = {...noLink,
      attributes: {
        ...noLink.attributes,
        get_absolute_url: '/absolute/',
      },
    }
    const propertyLink = {...noLink,
      attributes: {
        ...methodLink.attributes,
        __web_url__: '/web/',
      },
    }
    const testValues = [
      {},
      {value: input},
      {value: noAttrs},
      {value: noLink},
      {value: noLink, url: 'http://uptickhq.com'},
      {value: methodLink},
      {value: propertyLink},
    ]
    testValues.forEach(testValue => {
      snapshotTest(<ObjectLink {...testValue} />)
    })
    const objectLink = shallow(<ObjectLink value={noLink} a="b" renderer={Links} renderProps={{b: 'c'}} />)
    expect(objectLink.containsMatchingElement(<Links value={noLink} a="b" b="c" />)).toBe(true)
  })
})
