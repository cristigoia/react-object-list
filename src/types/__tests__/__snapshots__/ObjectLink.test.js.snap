// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectLink type snapshot tests 1`] = `
<span
  className="text-muted"
>
  -
</span>
`;

exports[`ObjectLink type snapshot tests 2`] = `
<span
  className="text-muted"
>
  -
</span>
`;

exports[`ObjectLink type snapshot tests 3`] = `
<div>
  <div>
    <div>
      TestModel 1
    </div>
  </div>
</div>
`;

exports[`ObjectLink type snapshot tests 4`] = `
<div>
  <div>
    <div>
      TestModel 1
    </div>
  </div>
</div>
`;

exports[`ObjectLink type snapshot tests 5`] = `
<a
  href="http://uptickhq.com"
>
  <div>
    <div>
      <div>
        TestModel 1
      </div>
    </div>
  </div>
</a>
`;

exports[`ObjectLink type snapshot tests 6`] = `
<div>
  <div>
    <div>
      TestModel 1
    </div>
  </div>
</div>
`;

exports[`ObjectLink type snapshot tests 7`] = `
<div>
  <div>
    <div>
      TestModel 1
    </div>
  </div>
</div>
`;
